### PR TITLE
ICS29: Add `fee transfer` command

### PR DESCRIPTION
**Pull request by romac**
Closes: #2714

See https://github.com/informalsystems/ibc-rs/pull/1981#discussion_r979819760

## Description

Add a `fee transfer` command which can be used to exercise the relayer incentivization support introduced in #1981.

### Testing

In order to test the `fee transfer` command, open 3 terminals:

__First terminal__

Setup the test chains

1. Run `nix shell github:informalsystems/cosmos.nix#ibc-go-v5-simapp` or `nix shell --extra-experimental-features nix-command --extra-experimental-features flakes github:informalsystems/cosmos.nix#ibc-go-v5-simapp`
2. `cd` to the `hermes` repo
3. Start the test chains with `CHAIN_COMMAND_PATH=simd cargo run --bin test_setup_with_fee_enabled_binary_channel`

__Second terminal__

Start a Hermes instance

1. Run `nix shell github:informalsystems/cosmos.nix#ibc-go-v5-simapp` or `nix shell --extra-experimental-features nix-command --extra-experimental-features flakes github:informalsystems/cosmos.nix#ibc-go-v5-simapp`
2. `cd` to the `hermes` repo
3. Run `source data/test/binary-channels.env`
4. Update `max_msg_num = 30` to `max_msg_num = 4` in the file `relayer-config.toml` located in `data/test/`
5. Run `CHAIN_COMMAND_PATH=simd cargo run --bin hermes -- --config $RELAYER_CONFIG start`

__Third terminal__

Transfer some tokens with fees. In addition this tests that if the pay message is not in the same batch as the transfer message, the relayer still receives the fees

1. `cd` to the `hermes` repo
2. Run `source data/test/binary-channels.env`
3. Check the balance of the relayers and users, in this case the only ones which will change are the relayer on chain A, user1 on chain A and user2 on chain B:
  - `gaiad --node $NODE_A_RPC_ADDR query bank balances $NODE_A_WALLETS_RELAYER_ADDRESS`
  - `gaiad --node $NODE_A_RPC_ADDR query bank balances $NODE_A_WALLETS_USER1_ADDRESS`
  - `gaiad --node $NODE_B_RPC_ADDR query bank balances $NODE_B_WALLETS_USER2_ADDRESS`
4. Transfer 30 times 5321 `samoleans` from `user1` to `user2`:
  -  `cargo run --bin hermes -- --config $RELAYER_CONFIG fee transfer --dst-chain $CHAIN_ID_B --src-chain $CHAIN_ID_A --src-port $PORT_A --src-channel $CHANNEL_ID_A --amount 5321 --denom $NODE_A_DENOM --receive-fee 1233 --ack-fee 772 --timeout-fee 2001 --timeout-seconds 120 --recipient $NODE_B_WALLETS_USER2_ADDRESS --key-name user1 --number-msgs 30`
5. Check the balances:
  - The relayer on chain A should have received 30 times 2005 `samoleans` (60150)
  - `user1` on chain A should have lost 30 times 2005+5321 `samoleans` (219780)
  - `user2` on chain B should have received 30 times 5321 `ibc/<hash>` (159630) 

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer-cli/src/cli_utils.rs [hunk 1]
```diff
@@ -102,3 +102,88 @@ pub fn spawn_chain_counterparty<Chain: ChainHandle>(
...
+    src_port_id: &PortId,
+    dst_chain_id: &ChainId,
+) -> Result<(), eyre::Report> {
+    use eyre::eyre;
+    use tracing::debug;
+
+    use ibc_relayer::chain::requests::{
+        IncludeProof, QueryChannelRequest, QueryClientStateRequest, QueryConnectionRequest,
+        QueryHeight,
+    };
```
Comment (seanchen1991) [inline crates/relayer-cli/src/cli_utils.rs:122]: It seems a bit non-standard to import these types at the function level. Is there any particular reason for doing it this way? If not, perhaps we should remain consistent and import these at the top of the file.

File: crates/relayer-cli/src/cli_utils.rs [hunk 2]
```diff
@@ -102,3 +102,88 @@ pub fn spawn_chain_counterparty<Chain: ChainHandle>(
         channel_connection_client,
     ))
 }
+
+/// Check that the relayer can send on the given channel and ensure that channels and chain identifiers match.
+/// To do this, fetch from the source chain the channel end, then the associated connection
+/// end, and then the underlying client state; finally, check that this client is verifying
+/// headers for the destination chain.
```
Comment (seanchen1991) [inline crates/relayer-cli/src/cli_utils.rs:109]: I think these comments would be more helpful as annotations in the body of the function.

File: crates/relayer-cli/src/cli_utils.rs (hunk 1 above)
Comment (seanchen1991) [reply crates/relayer-cli/src/cli_utils.rs]: [21de7d4](https://github.com/informalsystems/hermes/pull/2682/commits/21de7d431fcf574c42d50f018fc1a19e4d439cea)

**Review by seanchen1991 — APPROVED** [review]
I love the effort being put into comments and documentation 😄

**Comment by romac** [comment]
> love the effort being put into comments and documentation 😄

Yep, thanks @ljoss17 for the thorough testing instructions :)

**Pull request merged**